### ISSUE TITLE
Emoji Support

### ISSUE THREAD

**Issue by kedarv** [barryvdh/laravel-dompdf#382]
This is probably UTF related, but does anyone have a fix for getting emoji to be rendered properly?

**Comment by laurentiusar** [comment]
Any news regarding this situation?

**Comment by kedarv** [comment]
Nope, sorry! I just removed emoji from text as a bandaid solution

**Comment by laurentiusar** [comment]
Use a font that contains emoji and it works :)

**Comment by eckzhou** [comment]
@laurentiusar What font did you use? I tried OpenSanEmoji but seems it didn't work.

**Comment by Ramonivandy** [comment]
no update for this issue ?

**Comment by Ramonivandy** [comment]
i havent try this, but it should work 
https://groups.google.com/forum/#!topic/dompdf/UE2ISFHYYuE

**Comment by Ramonivandy** [comment]
> i havent try this, but it should work
> https://groups.google.com/forum/#!topic/dompdf/UE2ISFHYYuE

already try this and it work

> 
![image](https://user-images.githubusercontent.com/36778458/60175643-53a18780-983e-11e9-88b9-8e0dc503dc7d.png)

**Comment by simonschaufi** [comment]
Related: https://github.com/dompdf/dompdf/issues/1698

**Comment by simonschaufi** [comment]
Dear stale bot, this issue should still stay open. Thank you ;)

**Comment by davidcandreva** [comment]
+1

**Comment by cairocampos** [comment]
+1

**Comment by simonschaufi** [comment]
Dear stale bot, this issue should still stay open. Thank you ;)

**Comment by Jupitercouple29** [comment]
Any updates on this?
I displayed the emojis and it showed something like this.
`{"name":"message","value":"\ud83d\udca9\ud83d\udca9\ud83d\udca9\ud83d\udca9\ud83d\udca9\ud83d\udca9\ud83d\udca9\ud83d\udca9"}`

Still waiting for the solution.
Thanks!

**Comment by Jupitercouple29** [comment]
I'm working on laravel, I fixed this issue using this library
https://github.com/xuejd3/laravel-emoji

**Comment by simonschaufi** [comment]
Dear stale bot, this issue should still stay open. Thank you ;)

**Comment by simonschaufi** [comment]
.

**Comment by simonschaufi** [comment]
.

**Comment by JorisDebonnet** [comment]
This is [a limitation of dompdf](https://github.com/dompdf/dompdf/issues/1698), and so can/should not be fixed in this repo. I personally believe this issue should be closed.

**Comment by Jupitercouple29** [comment]
Have you guys tried this? https://github.com/xuejd3/laravel-emoji
I imported it on my composer.json. 
`"xuejd3/laravel-emoji": "^1.0",`

And I displayed it in this way. 
` <?php echo Emoji::toImage($property['value']) ?>`

Here, $property['value'] holds the emoji code. Please try this. I displayed emoji and even succeed to print it on PDF.

**Comment by JorisDebonnet** [comment]
> Have you guys tried this? https://github.com/xuejd3/laravel-emoji I imported it on my composer.json. `"xuejd3/laravel-emoji": "^1.0",`
> 
> And I displayed it in this way. ` <?php echo Emoji::toImage($property['value']) ?>`

I guess that is a viable workaround, but since it uses images, it's not a fix for all situations. So, indeed useful for some applications, but in general, a true fix for this issue can only come from dompdf (or one of its dependencies) ☺️

**Comment by simonschaufi** [comment]
.

**Comment by simonschaufi** [comment]
.

**Comment by simonschaufi** [comment]
.

**Comment by simonschaufi** [comment]
.

**Comment by DutchPrince** [comment]
.

**Comment by JorisDebonnet** [comment]
Emoji support does not depend on any work in this repository, but on dompdf itself. 

Related issues: 
https://github.com/dompdf/dompdf/issues/1698
https://github.com/dompdf/php-font-lib/issues/113 <== especially track this one
